codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,217 @@ public TextHeaders clear() {
+                return StringUtil.quoteCsv(value);
+            }
+        };
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence value) {
+            return addQuotedValue(name, StringUtil.quoteCsv(value));
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            return addQuotedValue(name, commaSeparate(charSequenceQuoter, values));
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            return addQuotedValue(name, commaSeparate(charSequenceQuoter, values));
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            return addQuotedValue(name, commaSeparate(objectQuoter, values));
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            return addQuotedValue(name, commaSeparate(objectQuoter, values));
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(charSequenceQuoter, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(charSequenceQuoter, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(objectQuoter, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(objectQuoter, values));
+            return DefaultTextHeaders.this;
+        }
+
+        private TextHeaders addQuotedValue(CharSequence name, CharSequence escapedValue) {
+            CharSequence currentValue = DefaultTextHeaders.super.get(name);
+            if (currentValue == null) {
+                DefaultTextHeaders.super.add(name, escapedValue);
+            } else {
+                if (!currentValue.equals(escapedValue)) {
+                    DefaultTextHeaders.super.set(name, commaSeparateQuotedValues(currentValue, escapedValue));
+                }
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        private <T> CharSequence commaSeparate(CsvValueQuoter<T> quoter, T... values) {
+            StringBuilder sb = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    sb.append(quoter.quote(values[i])).append(", ");
```

Comment:
I think we should just be appending a `,` character (no additional space).  WDYT?